src/packets.c

```diff
@@ -1281,11 +1289,226 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
                   break;
               }
               }
-              place_slab_type_on_map(slbkind, slab_subtile(subtile_slab_fast(stl_x), 0), slab_subtile(subtile_slab_fast(stl_y), 0), plyr_idx, 0);
+              place_slab_type_on_map(slbkind, slab_subtile(subtile_slab_fast(stl_x), 0), slab_subtile(subtile_slab_fast(stl_y), 0), get_selected_player_for_cheat(plyr_idx), 0);
+          }
+            unset_packet_control(pckt, PCtr_LBtnRelease);
+        }
+        break;
+    case PSt_LevelCreatureUp:
+            thing = get_creature_near(x, y);
+            if (!thing_is_creature(thing))
+            player->thing_under_hand = 0;
+        else
+            player->thing_under_hand = thing->index;
+        if (((pckt->control_flags & PCtr_LBtnRelease) != 0) && ((pckt->control_flags & PCtr_MapCoordsValid) != 0))
+        {
+             if (player->thing_under_hand > 0)
+          {
+                creature_increase_level(thing);
+          }
+        unset_packet_control(pckt, PCtr_LBtnRelease);    
+        }
+        break;
+    case PSt_KillPlayer:
+          i = get_selected_player_for_cheat(-1);
+          struct PlayerInfo* PlayerToKill = get_player(i);
+          if (player_exists(PlayerToKill))
+          {
+          thing = get_player_soul_container(PlayerToKill->id_number);
+          if (thing_is_dungeon_heart(thing))
+            {
+                thing->health = -1;
+            }
+          }
+        break;
+    case PSt_HeartHealth:
+          thing = get_player_soul_container(plyr_idx);
+            if (is_key_pressed(KC_ADD, KMod_ALT))
+            {
+                if (thing_is_dungeon_heart(thing))
+                    {
+                        thing->health++;
+                    }
+            }
+            else if (is_key_pressed(KC_EQUALS, KMod_SHIFT))
+            {
+                if (thing_is_dungeon_heart(thing))
+                    {
+                        thing->health++;
```

Comment:
I believe you want to remove health here right?

Reply:
> 
> 
> I believe you want to remove health here right?

Remove health where? There are keys for adding health, and keys for removing health.